src/main.rs

```diff
@@ -0,0 +1,254 @@
+use std::path::PathBuf;
+
+use clap::Parser;
+
+const DEFAULT_COMPRESSION_LEVEL: usize = 6;
+
+#[derive(Parser, Debug)]
+#[clap(
+    about = r"Compress or uncompress FILEs (by default, compress FILEs in-place).
+
+Mandatory arguments to long options are mandatory for short options too.",
+    version,
+    after_help = r"
+With no FILE, or when FILE is -, read standard input.
+
+Report bugs to <bug-gzip@gnu.org>."
+)]
+struct Args {
+    /// Convert end-of-lines to local OS conventions
+    #[clap(short, long)]
+    ascii: bool,
+
+    /// Output to stdout
+    #[clap(short = 'c', long = "stdout")]
+    to_stdout: bool,
+
+    /// Decompress
+    #[clap(short, long)]
+    decompress: bool,
+
+    /// Don't ask questions, compress links
+    #[clap(short, long)]
+    force: bool,
+
+    /// Keep (don't delete) input files
+    #[clap(short, long)]
+    keep: bool,
+
+    /// Don't save or restore the original file name
+    #[clap(short, long)]
+    no_name: bool,
+
+    /// Don't save or restore the original file time
+    #[clap(short = 'm')]
+    no_time: bool,
+
+    /// Save or restore the original modification time
+    #[clap(short = 'M', long)]
+    time: bool,
```

Comment:
The help messages are guarded by `UNDOCUMENTED` in gzip.c. Let's do `hide = true` for now.